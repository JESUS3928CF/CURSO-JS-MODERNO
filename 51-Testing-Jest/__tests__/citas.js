import Citas from "../js/classes/Citas.js";

describe("Probar la clase de citas", () => {

    //- La instanciamos para utilizarla en las pruebas 
    const citas = new Citas();

    /// declarando el id de forma global
    const id = Date.now();

    test('Agregar una nueva cita', () => { 
        const citaObj = {
            mascota: 'Hook',
            propietario: 'Jesús',
            telefono: '323333',
            fecha: '10-12-2020',
            hora: '10:30',
            sintomas: 'Solo duerme',
        };

        citaObj.id = id;

        //- Aca el método que vamos a probar

        citas.agregarCita(citaObj);

        //- Prueba
        expect(citas).toMatchSnapshot();
     })


     /// Segunda Prueba 
     test("Actualizar Cita", () => {
        //* Aca estamos actualizando esta cita
        const citaActualizada = {
            id, 
            mascota: 'Nuevo Nombre',
            propietario: 'Jesús',
            telefono: '323333',
            fecha: '10-12-2020',
            hora: '10:30',
            sintomas: 'Solo duerme',
        };

        citas.editarCita(citaActualizada);

        expect(citas).toMatchSnapshot();
     });

     /// Test de eliminar
     test("Eliminar cita", () => {
        citas.eliminarCita(id);

        //! Lo que esperamos 
        expect(citas).toMatchSnapshot();

     })
});