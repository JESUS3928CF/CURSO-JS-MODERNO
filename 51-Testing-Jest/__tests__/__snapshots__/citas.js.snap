// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Probar la clase de citas Actualizar Cita 1`] = `
Citas {
  "citas": [
    {
      "fecha": "10-12-2020",
      "hora": "10:30",
      "id": 1684039714648,
      "mascota": "Nuevo Nombre",
      "propietario": "Jesús",
      "sintomas": "Solo duerme",
      "telefono": "323333",
    },
  ],
}
`;

exports[`Probar la clase de citas Agregar una nueva cita 1`] = `
Citas {
  "citas": [
    {
      "fecha": "10-12-2020",
      "hora": "10:30",
      "id": 1684039714648,
      "mascota": "Hook",
      "propietario": "Jesús",
      "sintomas": "Solo duerme",
      "telefono": "323333",
    },
  ],
}
`;

exports[`Probar la clase de citas Eliminar cita 1`] = `
Citas {
  "citas": [],
}
`;
